app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsController.scala

```diff
@@ -264,95 +267,11 @@ object CheckClaimantDetailsController {
-  }
-
-  def extractContactAddress(fillingOutClaim: FillingOutClaim): Option[ContactAddress] = {
-    val draftC285Claim = fillingOutClaim.draftClaim.fold(identity)
-    draftC285Claim.mrnContactAddressAnswer
-      .orElse(
-        Applicative[Option]
-          .map2(draftC285Claim.displayDeclaration, draftC285Claim.declarantTypeAnswer) { (declaration, declarantType) =>
-            (declarantType match {
-              case DeclarantTypeAnswer.Importer | DeclarantTypeAnswer.AssociatedWithImporterCompany =>
-                declaration.displayResponseDetail.consigneeDetails.flatMap(_.contactDetails)
-              case DeclarantTypeAnswer.AssociatedWithRepresentativeCompany                          =>
-                declaration.displayResponseDetail.declarantDetails.contactDetails
-            }).flatMap { contactDetails =>
-              Applicative[Option].map2(contactDetails.addressLine1, contactDetails.postalCode) {
-                (addressLine1, postCode) =>
-                  ContactAddress(
-                    addressLine1,
-                    contactDetails.addressLine2,
-                    None,
-                    contactDetails.addressLine3.getOrElse(""),
-                    postCode,
-                    contactDetails.countryCode.map(Country(_)).getOrElse(Country.uk)
-                  )
-              }
-            }
-          }
-          .flatten
-      )
-  }
-
-  def removeContactDetails(fillingOutClaim: FillingOutClaim): FillingOutClaim = {
-    val draftC285Claim     = fillingOutClaim.draftClaim.fold(identity)
-    val displayDeclaration = draftC285Claim.displayDeclaration.map(declaration =>
-      declaration.copy(displayResponseDetail =
-        declaration.displayResponseDetail.copy(
-          consigneeDetails =
-            declaration.displayResponseDetail.consigneeDetails.map(consignee => consignee.copy(contactDetails = None)),
-          declarantDetails = declaration.displayResponseDetail.declarantDetails.copy(contactDetails = None)
-        )
-      )
-    )
-    fillingOutClaim.copy(draftClaim =
-      draftC285Claim
-        .copy(mrnContactDetailsAnswer = None, mrnContactAddressAnswer = None, displayDeclaration = displayDeclaration)
-    )
-  }
-
   def isMandatoryDataAvailable(fillingOutClaim: FillingOutClaim): Boolean = {
     val draftC285Claim = fillingOutClaim.draftClaim.fold(identity)
-    val contactDetails = extractContactDetails(fillingOutClaim)
-    val contactAddress = extractContactAddress(fillingOutClaim)
-
-    if (draftC285Claim.mrnContactDetailsAnswer.isDefined && draftC285Claim.mrnContactAddressAnswer.isDefined)
-      true
-    else if (
-      contactDetails.name.isDefined &&
-      contactDetails.email.isDefined &&
-      contactAddress.map(_.line1).isDefined &&
-      contactAddress.map(_.postcode).isDefined
-    )
-      true
-    else
-      false
+    (draftC285Claim.mrnContactDetailsAnswer, draftC285Claim.mrnContactAddressAnswer)
+      .mapN((_, _) => true)
```

Comment:
Looks good, I wanted to tell you about this that is a better alternative syntax over Applicative[Option].map2(....)

Reply:
Thanks